internal/cache/util/util_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/data"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	. "github.com/jacobsa/ogletest"
+)
+
+func TestUtil(t *testing.T) { RunTests(t) }
+
+type utilTest struct {
+	fileSpec data.FileSpec
+	flag     int
+	uid      int
+	gid      int
+}
+
+const FileDir = "/some/dir/"
+const FileName = "foo.txt"
+
+func init() { RegisterTestSuite(&utilTest{}) }
+
+func (ut *utilTest) SetUp(*TestInfo) {
+	operations.RemoveDir(FileDir)
+	ut.flag = os.O_RDWR
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(fmt.Errorf("error while finding home directory: %v", err))
```

Comment:
Can we use t.Fatal() or t.FailNow() instead of panic()?
Ref - https://blog.devgenius.io/write-better-golang-code-by-avoiding-panic-in-tests-f6a7a22937e6#:~:text=So%2C%20to%20summarize%2C%20you',to%20writing%20better%20Go%20code.

Reply:
I also thought of using that but I couldn't find how a similar option in https://github.com/jacobsa/ogletest. Let me know if you know of any similar way.

Reply:
Ohh, we can't take *testing.T as input in SetUp method as per the current implementation. Then we don't have any option other than panic().